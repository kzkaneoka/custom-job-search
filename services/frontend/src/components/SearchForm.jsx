import React from "react";
import PropTypes from "prop-types";
import { Formik } from "formik";
import * as Yup from "yup";

import "./form.css";

const SearchForm = props => {
  return (
    <div>
      <h1 className="title is-1">Job Search</h1>
      <hr />
      <br />
      <Formik
        initialValues={{
          words: "",
          location: ""
        }}
        onSubmit={(values, { setSubmitting, resetForm }) => {
          props.handleSearchFormSubmit(values);
          setSubmitting(false);
        }}
        validationSchema={Yup.object().shape({
          words: Yup.string().required(
            "At leaset one word is required. You can add multiple words separeated by comma."
          ),
          location: Yup.string().required("Location is required.")
        })}
      >
        {props => {
          const {
            values,
            touched,
            errors,
            isSubmitting,
            handleChange,
            handleBlur,
            handleSubmit
          } = props;
          return (
            <form onSubmit={handleSubmit}>
              <div className="field">
                <label className="label" htmlFor="input-words">
                  Words
                </label>
                <input
                  name="words"
                  id="input-words"
                  className={
                    errors.words && touched.words ? "input error" : "input"
                  }
                  type="text"
                  placeholder="Enter words. You can add multiple words separeated by comma."
                  value={values.words}
                  onChange={handleChange}
                  onBlur={handleBlur}
                />
                {errors.words && touched.words && (
                  <div className="input-feedback" data-testid="errors-words">
                    {errors.words}
                  </div>
                )}
              </div>
              <div className="field">
                <label className="label" htmlFor="input-location">
                  Location
                </label>
                <input
                  name="location"
                  id="input-location"
                  className={
                    errors.location && touched.location
                      ? "input error"
                      : "input"
                  }
                  type="text"
                  placeholder="Enter location."
                  value={values.location}
                  onChange={handleChange}
                  onBlur={handleBlur}
                />
                {errors.location && touched.location && (
                  <div className="input-feedback" data-testid="errors-location">
                    {errors.location}
                  </div>
                )}
              </div>
              <input
                type="submit"
                className="button is-primary"
                value="Search Jobs"
                disabled={isSubmitting}
              />
            </form>
          );
        }}
      </Formik>
    </div>
  );
};

export default SearchForm;

SearchForm.propTypes = {
  handleSearchFormSubmit: PropTypes.func.isRequired
};
