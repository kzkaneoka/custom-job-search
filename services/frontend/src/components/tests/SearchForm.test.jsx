import React from "react";
import { cleanup, fireEvent, wait } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import SearchForm from "../SearchForm";

afterEach(cleanup);

describe("renders", () => {
  const props = {
    handleSearchFormSubmit: () => {
      return true;
    }
  };

  it("properly", () => {
    const { getByText } = renderWithRouter(<SearchForm {...props} />);
    expect(getByText("Job Search")).toHaveClass("title");
  });

  it("default props", () => {
    const { getByLabelText, getByText } = renderWithRouter(
      <SearchForm {...props} />
    );

    const wordsInput = getByLabelText("Words");
    expect(wordsInput).toHaveAttribute("type", "text");
    expect(wordsInput).not.toHaveValue();

    const locationsInput = getByLabelText("Locations");
    expect(locationsInput).toHaveAttribute("type", "text");
    expect(locationsInput).not.toHaveValue();

    const buttonInput = getByText("Submit");
    expect(buttonInput).toHaveValue("Submit");
  });

  it("a snapshot properly", () => {
    const { asFragment } = renderWithRouter(<SearchForm {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
