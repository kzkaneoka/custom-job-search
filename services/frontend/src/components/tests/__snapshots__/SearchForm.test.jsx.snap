// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot properly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title is-1"
    >
      Job Search
    </h1>
    <hr />
    <br />
    <form>
      <div
        class="field"
      >
        <label
          class="label"
          for="input-words"
        >
          Words
        </label>
        <input
          class="input"
          id="input-words"
          name="words"
          placeholder="Enter words. You can add multiple words separeated by comma."
          type="text"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="input-locations"
        >
          Locations
        </label>
        <input
          class="input"
          id="input-locations"
          name="locations"
          placeholder="Enter locations. You can add multiple locations separeated by comma."
          type="text"
          value=""
        />
      </div>
      <input
        class="button is-primary"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
